homeassistant/components/samsungtv/__init__.py

```diff
@@ -140,13 +141,17 @@ async def _async_create_bridge_with_updated_data(hass, entry):
 
     bridge = _async_get_device_bridge({**entry.data, **updated_data})
 
-    if not entry.data.get(CONF_MAC) and bridge.method == METHOD_WEBSOCKET:
+    mac = entry.data.get(CONF_MAC)
+    if not mac and bridge.method == METHOD_WEBSOCKET:
         if info:
             mac = mac_from_device_info(info)
         else:
             mac = await hass.async_add_executor_job(bridge.mac_from_device)
-        if mac:
-            updated_data[CONF_MAC] = mac
+
+    if not mac:
+        mac = getmac.get_mac_address(entry.data.get(CONF_HOST))
```

Comment:
I think this is ok since we aren't using it for the unique id, but it needs to run in the executor since get_mac_address can do I/O

Reply:
Done